infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/brokerphases/BrokerEnvironmentFactory.java

```diff
@@ -153,14 +153,22 @@ private String generateUniqueName(String suffix) {
     return NameGenerator.generate(suffix, 6);
   }
 
+  @VisibleForTesting
+  protected String generateContainerNameFromImageRef(String image) {
+    return image.toLowerCase().replaceAll("[^/]*/", "").replaceAll("[^\\d\\w-]", "-");
+  }
+
   private Container newContainer(
       RuntimeIdentity runtimeId,
       List<EnvVar> envVars,
       String image,
       @Nullable String brokerVolumeName) {
+    // There's a chance the full image reference may be over the name limit of 63 chars
+    // so we need to remove registry hostname
```

Comment:
@amisevsk this comment looks obsolete

Reply:
or maybe it meant to be added to the `generateContainerNameFromImageRef` doc?

Reply:
Moved docs to `generateContainerNameFromImageRef`